packages/framework/ai-collab/src/explicit-strategy/index.ts

```diff
@@ -0,0 +1,358 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	getSimpleSchema,
+	Tree,
+	type SimpleTreeSchema,
+	type TreeNode,
+} from "@fluidframework/tree/internal";
+// eslint-disable-next-line import/no-internal-modules
+import { zodResponseFormat } from "openai/helpers/zod";
+import type {
+	ChatCompletionCreateParams,
+	// eslint-disable-next-line import/no-internal-modules
+} from "openai/resources/index.mjs";
+import { z } from "zod";
+
+import type { OpenAiClientOptions, TokenLimits, TokenUsage } from "../aiCollabApi.js";
+
+import { applyAgentEdit } from "./agentEditReducer.js";
+import type { EditWrapper, TreeEdit } from "./agentEditTypes.js";
+import { IdGenerator } from "./idGenerator.js";
+import {
+	getEditingSystemPrompt,
+	getPlanningSystemPrompt,
+	getReviewSystemPrompt,
+	toDecoratedJson,
+	type EditLog,
+} from "./promptGeneration.js";
+import { generateGenericEditTypes } from "./typeGeneration.js";
+import { fail } from "./utils.js";
+
+const DEBUG_LOG: string[] = [];
+
+/**
+ * {@link generateTreeEdits} options.
+ *
+ * @internal
+ */
+export interface GenerateTreeEditsOptions {
```

Comment:
Feels somewhat redundant with another type I saw that had a lot of these same things. Either composition or inheritance might be worth exploring here. Same with the two response interfaces below.